auto_cpufreq/battery_scripts/shared.py

```diff
@@ -0,0 +1,126 @@
+            )
+            return False
+        else:
+            return True
+
+    def get_battery_paths(self, bat, mode):
+        paths = None
+        if mode == "start":
+            paths = self.start_paths[bat]
+        else:
+            paths = self.stop_paths[bat]
+        return paths
+
+    def set_battery(self, bat, mode, value) -> bool:
+        """
+        Return true/false depending on if command is executed and fails (or succeeds)
+        """
+        paths = self.get_battery_paths(bat, mode)
+
+        for p in paths:
+            if os.path.isfile(p):
+                try:
+                    check_output(f"echo {value} | tee {p}", shell=True, text=True)
+                    return True
+                except CalledProcessError as e:
+                    print(
+                        f"WARNING: Could not write battery threshold {value} to {p}: {e.output}"
+                    )
+                    return False
+            else:
+                print(f"WARNING: {p} does NOT exist")
+        return True
+
+    def get_current_threshold(self, bat, mode):
+        paths = self.get_battery_paths(bat, mode)
+        for p in paths:
+            if os.path.isfile(p):
+                try:
+                    return check_output(
+                        ["cat", POWER_SUPPLY_DIR + bat + f"/charge_{mode}_threshold"]
+                    )
+                except CalledProcessError as e:
+                    print(
+                        f"ERROR: failed to read battery {bat} {mode} threshold in file: {p}: {e.output}"
+                    )
+        raise OSError(f"ERROR: failed to get any battery threshold for {bat}, {mode}!")
+
+    def print_thresholds(self):
+        print(
+            "\n-------------------------------- Battery Info ---------------------------------\n"
+        )
+        print(f"battery count = {len(self.batteries)}")
+        for bat in self.batteries:
+            try:
+                print(
+                    bat, "start threshold =", self.get_current_threshold("start", bat)
+                )
+                print(bat, "stop threshold =", self.get_current_threshold("stop", bat))
+            except Exception as e:
+                print(f"ERROR: failed to read battery {bat} thresholds:", repr(e))
+
+    def setup(self):
+        if not (
+            self.config.has_option("battery", "enable_thresholds")
+            and self.config["battery"]["enable_thresholds"] == "true"
+        ):
+            return
+        elif not os.path.exists(POWER_SUPPLY_DIR):
+            print(f"WARNIN: {POWER_SUPPLY_DIR} does NOT exist")
+            return
+
+        batteries = [
+            name for name in os.listdir(POWER_SUPPLY_DIR) if name.startswith("BAT")
+        ]
+
+        for bat in batteries:
+            # First set 0 for start and 100 for stop, sometimes we can have conflicting values otherwise which will result in an 'invalid argument' upon writing.
```

Comment:
It's probably good idea to reflect some of these comments as [part of battery charging thresholds docs for this feature.](https://github.com/AdnanHodzic/auto-cpufreq/#battery-charging-thresholds:~:text=start_threshold%20%3D%2020%20(should,60%25%20or%2080%25))